common/unsafe/src/main/java/org/apache/spark/unsafe/Platform.java

```diff
@@ -51,6 +51,8 @@
       // Since java.nio.Bits.unaligned() doesn't return true on ppc (See JDK-8165231), but
       // ppc64 and ppc64le support it
       _unaligned = true;
+    } else if (arch.matches("s390x")) {
```

Comment:
`.contains`? that isn't a regex. You can put it into the clause above and modify the comment

Reply:
This can still be changed

Reply:
@srowen  : did not get your <.contains> comment what exactly you are trying to say
I have intentionally written a else if blk to keep processor specific changes in seperate block.

are you asking me to add "s390x" in the same "ppc64le" blk ... let me know

Reply:
Yes, keep them together. And it does not need to be a regex with `matches`; just use `contains`.

Reply:
> @ketank-new Could you address the comments and resolve the conflicts?

sure